notifications_utils/recipients.py

```diff
@@ -130,6 +134,21 @@ def allowed_to_send_to(self):
             return True
         return all(allowed_to_send_to(row.recipient, self.guestlist) for row in self.rows)
 
+    @cached_property
+    def international_sms_count(self):
+        if self.template_type != "sms":
+            return 0
+        return sum(self._international_sms_count_generator())
+
+    def _international_sms_count_generator(self):
```

Comment:
isn't having a separate generator much less efficient than getting the count from errors function? That is - we now have to check each phone number twice - once for the other errors, and once to check the international limit, when it could all be done in one go?

Reply:
> isn't having a separate generator much less efficient than getting the count from errors function? That is - we now have to check each phone number twice - once for the other errors, and once to check the international limit, when it could all be done in one go?

I think the generator could work better because this evaluation for international sms would be independent of the other error evaluations. 
Also it has the benefit of lazy evaluation which could also be beneficial because we are potentially dealing with a large data set and it would be more memory efficient.

Reply:
Wow, amazing, well done Chris! And also the magic of teamwork 😁